azext_edge/tests/edge/init/int/opcua_helper.py

```diff
@@ -0,0 +1,33 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from time import sleep
+from typing import Optional
+from ....helpers import run
+
+# args to wait for --simulate-plc
+MAX_TRIES = 10
+SLEEP_INTERVAL = 60
+
+
+def assert_simulate_plc_args(
+    custom_location: str,
+    simulate_plc: Optional[bool] = None,
+    **_
+):
+    if not simulate_plc:
+        simulate_plc = False
+    # note that the simulator may take a bit
+    query_result = run(f"az iot ops asset query --cl {custom_location}")
+    tries = 0
+    while simulate_plc and not query_result and tries < MAX_TRIES:
```

Comment:
To test --simulate-plc, we can do cluster side check the opcplc-00000 pod was deployed and related asset & aep CRs are created. I don't see why we need to test resource hydration via rsync rules.

Reply:
i thought we wanted to avoid cluster side if there is a cloud side solution...

Reply:
The goal is to assert cloud side when resources are created cloud first, then assert cluster side when resources are created edge first. Both the simulator and its asset & aep are edge side creation and rely on RSRs to hydrate cloud.

Reply:
changed to just asset the opcplc pod (rather than wait for the simulator to be done and have the assets/etc be populated).

Reply:
Sounds good